dbcon/joblist/tuplehashjoin.cpp

```diff
@@ -1863,19 +1863,25 @@ void TupleHashJoinStep::generateJoinResultSet(const vector<vector<Row::Pointer>
             {
                 uint32_t dbRoot = l_outputRG.getDBRoot();
                 uint64_t baseRid = l_outputRG.getBaseRid();
-                outputData->push_back(rgData);
+                outputData.push_back(rgData);
+                // Count the memory
+                if (UNLIKELY(!getMemory(l_outputRG.getMaxDataSize())))
```

Comment:
Sorry what do we do here? If i read the semantics right we just drop some of the data w/o an explicit exception.

Reply:
In this case, we're building a vector of RGData, max 8192 rows each RGData. All we do here is put a throttle on how big that vector can grow based on memory usage.
The full text is 
```
                if (UNLIKELY(!getMemory(l_outputRG.getMaxDataSize())))
                {
                    // Don't let the join results buffer get out of control.
                    RowGroup out(outputRG); // Useless line. Going away.
                    sendResult(outputData);
                    outputData.clear();
                    returnMemory();
                }

```
So if we fail to get memory, sendResult, clear the buffer and return the memory. The next time 'round starts with an empty vector. SendResult() iterates the vector and inserts into DL. No loss of data.